proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,366 @@
+For other settings not in config section, we have/will have other commands. E.g. for [trustedSigners section](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#trustedsigners-section), we have [dotnet nuget trust](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-trust) command.
+
+#### Arguments
+
+- CONFIG_KEY
+
+Specify the key of the settings that are to be removed.
+
+If the specified `CONFIG_KEY` is not a key in [config section](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#config-section), a message is displayed indicating the `CONFIG_KEY` could not be unset.
+
+#### Options
+
+- --configfile <FILE>
+
+The NuGet configuration file (nuget.config) to use. If specified, only the settings from this file will be used. If it's not specified, `%AppData%\NuGet\NuGet.Config` (Windows), or `~/.nuget/NuGet/NuGet.Config` or `~/.config/NuGet/NuGet.Config` (Mac/Linux) is used. See [On Mac/Linux, the user-level config file location varies by tooling.](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#on-maclinux-the-user-level-config-file-location-varies-by-tooling)
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+#### Examples
+
+- Unset the `signatureValidationMode` in the user-wide NuGet configuration file.
+
+```
+dotnet nuget config unset signatureValidationMode
+
+```
+
+- Unset the `defaultPushSource` in the specified NuGet configuration file.
+
+```
+dotnet nuget config unset defaultPushSource --configfile C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+
+```
+
+## Future Work
+1. The `dotnet nuget config path/get` is a community ask. We will discuss if adding this command into NuGet.exe CLI, in the future.
+
+2. NuGet.exe [config command](https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-config) is implemented. 
+But the behavior is confusing: the `set` command will set the property which appears last when loading(for all writable files), so it's not updating the closest NuGet configuration file, but the user-wide NuGet configuration file.(Related code: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs#L229)
+We keep the behavior of `dotnet nuget config set` the same with above for now.
+But we might need to change this behavior in the future. 
+Considering this will be a breaking change, we will only consider doing it in main version change, if there are enough votes.
+
+## Open Questions
+1. Shall we use `dotnet nuget config path` or `dotnet nuget config paths` to get all configuration file paths?
+
+2. `WORKING_DIRECTORY` is changed from an option to an argument. So we have `dotnet nuget config path <WORKING_DIRECTORY>` and `dotnet nuget config get <ALL|CONFIG_KEY> <WORKING_DIRECTORY>`. Is it okey if we have two arguments in the second command? And if it's key, only the second argument could be optional, right? 
```

Comment:
This seems OK to me, but I'd like to hear from and defer to @baronfel, as he's the expert.

Reply:
Hi @baronfel , could you help with this open question? Thanks!

Reply:
Good eye on the potential for ambiguity of the grammar here. Let me make sure I have the proposed grammar correct:

> dotnet nuget config get <all|CONFIG_KEY> [<WORKING_DIRECTORY>]

meaning 
> the first argument is required, and will either be the string 'all' or the name of an existing config key. the second key is optional and if present represents the working directory to locate the nuget config file from

Is that correct? if so, **I see no problem and am ok with the proposal**. in cases where there is some potential for ambiguity, consider adding a flag to allow for explicit disambiguation (i.e. `-w <path>` like we rejected earlier), but we've already had that conversation so I don't think we need to revisit it.

Reply:
Yep, that's correct :)
Great! Thanks for the confirmation! I moved this open question to Considerations.